src/Microsoft.Health.Fhir.Api/Features/Throttling/ThrottlingMiddleware.cs

```diff
@@ -267,6 +286,27 @@ private async Task Return429(HttpContext context)
             await context.Response.Body.WriteAsync(_throttledBody);
         }
 
+        private async Task Return429FromRequestRateExceededException(RequestRateExceededException exception, HttpContext context)
+        {
+            _logger.LogWarning($"Returning 429 from unhandled {nameof(RequestRateExceededException)}");
+
+            context.Response.StatusCode = StatusCodes.Status429TooManyRequests;
+
+            if (exception.RetryAfter.HasValue)
+            {
+                context.Response.Headers.AddRetryAfterHeaders(exception.RetryAfter.Value);
```

Comment:
If the exception doesn't have a retry after value, should we add one for consistency?

Reply:
I guess we can...